src/util/methodParameters.ts

```diff
@@ -52,11 +56,71 @@ export function buildTrade<TTradeType extends TradeType>(
   );
 
   // TODO: populate v4Routes
-  const v4Routes: {
-    routev4: V4RouteRaw<Currency, Currency>;
-    inputAmount: SDKCurrentAmount<Currency>;
-    outputAmount: SDKCurrentAmount<Currency>;
-  }[] = [];
+  const v4Routes = _.map<
```

Comment:
{v4|v3|v2}routes logic could be simplified with some common interfaces as all of them seem to do same thing, but not blocking for now. maybe a todo in the future.